client/src/sections/courses/create/InitialSection.tsx

```diff
@@ -23,9 +24,10 @@ interface InitialSectionProps {
 }
 
 const InitialSection: FC<InitialSectionProps> = ({ register, errors, control }) => {
+  const [category, setCategory] = useState([]);
```

Comment:
We do not need another state for saving category, please get the category from rtk query itself down below line 40 and use it in line 30